src/auth/auth.service.ts

```diff
@@ -2,31 +2,112 @@ import { Injectable } from '@nestjs/common';
 import { JwtService } from '@nestjs/jwt';
 import { UserService } from '../user/user.service';
 import { CryptoService } from '../utils/crypto.service';
-import { LoginDto } from './dto/auth';
+import { TokenDto } from './dto/auth';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserSession } from './entities/user-session.entity';
+import * as dayjs from 'dayjs';
+import { RefreshedToken, UserIdentifier } from './types/auth';
 
 @Injectable()
 export class AuthService {
   constructor(
     private readonly userService: UserService,
     private readonly jwtService: JwtService,
     private readonly hashService: CryptoService,
+    @InjectRepository(UserSession)
+    private readonly sessionRepository: Repository<UserSession>,
   ) {}
 
-  // Todo: typehint this? but how?
-  async validateUser(email: string, password: string): Promise<any> {
+  async validateUser(
+    email: string,
+    password: string,
+  ): Promise<UserIdentifier | null> {
     const user = await this.userService.findOne(email);
 
     if (user && (await this.hashService.compare(password, user.password))) {
```

Comment:
I propose to save the result of hashService.compare in a variable. Calling functions in an if statement is kind of ugly 😅

Reply:
Fixed in #49